src/lib/git/GitBackend.ts

```diff
@@ -1,10 +1,11 @@
 import path from 'path';
 import { EncryptedFS } from 'encryptedfs';
+import { git } from '../../../proto/js/Git';
 import { PassThrough } from 'readable-stream';
-import VaultManager from '../vaults/VaultManager';
 import uploadPack from './upload-pack/uploadPack';
 import GitSideBand from './side-band/GitSideBand';
 import packObjects from './pack-objects/packObjects';
+const { GitMessage, GitMessageType, InfoReply, InfoRequest, PackReply, PackRequest, VaultNamesReply } = git;
```

Comment:
Does this mean you are using GRPC as a transport for git? If not, the git code should have no idea about GRPC/protobuf types at all, that interface level code should be completely elided by the time you're in the core datastructures/models of Polykey.

Reply:
Core data structures and core models should be using domain specific objects and classes. In TS you can just create types like:

```
type X = {
  ...
};
```

Just like we are doing for Introcept.